src/mongo/delegates/SearchIndexer.class.php

```diff
@@ -44,13 +46,7 @@ public function __construct(\Tripod\Mongo\Driver $tripod, $readPreference = Read
         $this->labeller = new Labeller();
         $this->stat = $tripod->getStat();
         $this->config = $this->getConfigInstance();
-        $provider = $this->config->getSearchProviderClassName($this->tripod->getStoreName());
-
-        if (class_exists($provider)) {
-            $this->configuredProvider = new $provider($this->tripod);
-        } else {
-            throw new \Tripod\Exceptions\SearchException("Did not recognise Search Provider, or could not find class: $provider");
-        }
+        $this->setSearchProvider($this->tripod, $this->config);
```

Comment:
There is a mismatch in what is being passed around.

`setSearchProvider` expects the 2nd param to be a `Config`, but `$this->config` is an `IConfigInstance`. I think it's just a docblock error but it would be good to resolve